lang/cs/Org.Apache.REEF.Tests/Functional/Messaging/MessageTask.cs

```diff
@@ -59,44 +69,33 @@ public Optional<TaskMessage> Message
 
         public byte[] Call(byte[] memento)
         {
-            Console.WriteLine("Hello, CLR TaskMsg!");
-            Thread.Sleep(5 * 1000);
-            return null;
-        }
+            while (!(_messageManager.IsTaskMessageSent && _messageManager.IsContextMessageSent))
+            {
+                // Do not exit until task and context messages are both sent.
+                Thread.Sleep(5000);
```

Comment:
This waits at most 5 seconds after the last message is sent, right? Can this be done with an interrupt instead? Such that we don't delay test completion by so much?